base/applications/taskmgr/procpage.c

```diff
@@ -931,3 +933,156 @@ int CALLBACK ProcessPageCompareFunc(LPARAM lParam1, LPARAM lParam2, LPARAM lPara
     }
     return ret;
 }
+
+static DWORD _GetModuleFileNameExW(HANDLE hProcess, HMODULE hModule, LPWSTR lpFilename, DWORD nSize)
+{
+    typedef DWORD (WINAPI *PFN_GETMODULEFILENAMEEXW)(HANDLE, HMODULE, LPWSTR, DWORD);
+
+    HMODULE hLibrary;
+    PFN_GETMODULEFILENAMEEXW func;
+    DWORD dwRet = 0;
+
+    hLibrary = LoadLibraryW(L"psapi.dll");
+
+    if (hLibrary)
+    {
+        func = (PFN_GETMODULEFILENAMEEXW)GetProcAddress(hLibrary, "GetModuleFileNameExW");
+
+        if (func)
+        {
+            dwRet = func(hProcess, hModule, lpFilename, nSize);
+        }
+
+        FreeLibrary(hLibrary);
+    }
+
+    return dwRet;
+}
+
+static BOOL GetProcessExecutablePath(DWORD dwProcessId, LPWSTR lpExePath, DWORD dwLength)
+{
+    WCHAR *pszExePath;
+    HANDLE hProcess;
+    BOOL bSuccess = FALSE;
+
+    if (dwProcessId == 0)
+    {
+        return FALSE;
+    }
+    
+    pszExePath = HeapAlloc(GetProcessHeap(), 0, dwLength * sizeof(WCHAR));
+
+    if (!pszExePath)
+    {
+        return FALSE;
+    }
+    
+    /* PID = 4 or "System" */
+    if (dwProcessId == 4)
+    {
+        static const WCHAR szKernelPath[] = L"\\SystemRoot\\System32\\ntoskrnl.exe";
+
+        if (dwLength >= _countof(szKernelPath))
+        {
+            memcpy(pszExePath, szKernelPath, sizeof(szKernelPath));
+
+            bSuccess = TRUE;
+        }
+    }
+    else
+    {
+        hProcess = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, dwProcessId);
+
+        if (hProcess)
+        {
+            bSuccess = (_GetModuleFileNameExW(hProcess, NULL, pszExePath, dwLength) != 0);
+
+            CloseHandle(hProcess);
+        }
+    }
+
+    if (bSuccess)
+    {
+        /* Remove or replace path prefix if present */
+        if (wcsncmp(pszExePath, L"\\??\\", 4) == 0)
+        {
+            /* Remove "\??\" prefix */
+            size_t NewLen = wcslen(pszExePath) - 4;
+            memmove(pszExePath, pszExePath + 4, (NewLen + 1) * sizeof(WCHAR));
+        }
+        else if (_wcsnicmp(pszExePath, L"\\SystemRoot", 11) == 0)
+        {
+            WCHAR szWinDir[MAX_PATH];
+            HRESULT hr;
+
+            bSuccess = FALSE;
+            hr = SHGetFolderPathW(NULL, CSIDL_WINDOWS, NULL, 0, szWinDir);
```

Comment:
Please use NT path functions to deal with all that.

Reply:
Fixed.